cmd/skywire-visor/commands/root.go

```diff
@@ -581,3 +446,144 @@ func checkHvIsRunning(addr string, retries int) bool {
 	}
 	return false
 }
+
+func updateConfig(mLog *logging.MasterLogger, conf *visorconfig.V1) {
+	log := mLog.PackageLogger("visor:update-config")
+
+	log.Error("config version does not match visor version")
+	log.WithField("skywire version: ", visorBuildInfo.Version).Error()
+	var updstr string
+	match := strings.Contains("/tmp/", skywire)
+	log.Info("match:", match)
+	if match {
+		if _, err := os.Stat("cmd/skywire-cli/skywire-cli.go"); err == nil {
+			updstr = "go run cmd/skywire-cli/skywire-cli.go config gen -b"
+		}
+		log.Info("updstr:", updstr)
+	}
+	if updstr == "" {
+		updstr = "skywire-cli config gen -b"
+	}
+	if conf.Hypervisor != nil {
+		updstr = updstr + "i"
+	}
+	for _, j := range conf.Hypervisors {
+		if fmt.Sprintf("\t%s\n", j) != "" {
+			updstr = updstr + "x"
+			break
+		}
+	}
+	pkgenv := strings.Contains("/opt/skywire/apps", conf.Launcher.BinPath)
+	if pkgenv {
+		updstr = updstr + "p"
+	}
+	//there is no config *file* with stdin
+	if confPath != visorconfig.StdinName {
+		if _, err := exec.LookPath("stat"); err == nil {
+			if owner, err := script.Exec(`stat -c '%U' ` + confPath).String(); err == nil {
+				if (owner == "root") || (owner == "root\n") {
+					updstr = "sudo " + updstr
+				}
+			}
+		}
+		updstr = "\n		" + updstr + "ro " + confPath + "\n"
+	} else {
+		updstr = "\n		" + updstr + "n" + " | go run cmd/skywire-visor/skywire-visor.go -n"
+		if launchBrowser {
+			updstr = updstr + "b"
+		}
+		updstr = updstr + "\n"
+	}
+	updstr = "\n		" + updstr + "\n"
+	log.Info("please update your config with the following command:\n", updstr)
+	log.Fatal("failed to start skywire")
+
+}
+
+func logBuildInfo(mLog *logging.MasterLogger) {
+	log := mLog.PackageLogger("buildinfo")
+	var fork string
+	var branch string
+	var nocommit string
+	//indicates how skywire was started
+	skywire = os.Args[0]
+	//indicates where skywire was started
+	path, err := os.Getwd()
+	if err != nil {
+		log.WithError(err).Fatal()
+	}
+	workDir = path
+	//retrieve build info
+	visorBuildInfo = buildinfo.Get()
+	if visorBuildInfo.Version == "unknown" {
```

Comment:
About the following blocks of code: 

What is the value of checking for all these detailed build infos? If we really do need them, should it be moved to the buildinfo pkg?